test/uk/gov/hmrc/merchandiseinbaggage/controllers/PreviousDeclarationDetailsControllerSpec.scala

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.controllers
+
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.merchandiseinbaggage.connectors.MibConnector
+import uk.gov.hmrc.merchandiseinbaggage.model.api.{Amendment, CategoryQuantityOfGoods, Country, Currency, DeclarationGoods, DeclarationId, DeclarationType, ExportGoods, GoodsVatRates, ImportGoods, JourneyDetails, JourneyInSmallVehicle, NotRequired, Paid, PaymentStatus, Port, PurchaseDetails, SessionId, YesNoDontKnow}
+import uk.gov.hmrc.merchandiseinbaggage.model.core.DeclarationJourney
+import uk.gov.hmrc.merchandiseinbaggage.stubs.MibBackendStub.givenPersistedDeclarationIsFound
+import uk.gov.hmrc.merchandiseinbaggage.views.html.{PreviousDeclarationDetailsView}
+import uk.gov.hmrc.merchandiseinbaggage.wiremock.WireMockSupport
+import uk.gov.hmrc.merchandiseinbaggage.config.MibConfiguration
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import java.time.{LocalDate, LocalDateTime}
+
+class PreviousDeclarationDetailsControllerSpec extends DeclarationJourneyControllerSpec with WireMockSupport with MibConfiguration {
+
+  val journeyDetails: JourneyDetails = JourneyInSmallVehicle(
+    Port("DVR", "title.dover", isGB = true, List("Port of Dover")),
+    LocalDate.now(),
+    "T5 RRY"
+  )
+
+  def createImport(desc: String, price: String): ImportGoods = ImportGoods(
+    CategoryQuantityOfGoods(desc, "1"),
+    GoodsVatRates.Twenty,
+    YesNoDontKnow.Yes,
+    PurchaseDetails(price, Currency("EUR", "title.euro_eur", Some("EUR"), List("Europe", "European")))
+  )
+
+  def createExport(desc: String, price: String): ExportGoods = ExportGoods(
+    CategoryQuantityOfGoods(desc, "1"),
+    Country("FR", "title.france", "FR", isEu = true, Nil),
+    PurchaseDetails(price, Currency("EUR", "title.euro_eur", Some("EUR"), List("Europe", "European")))
+  )
+
+  def createAmendment(desc: String, paymentStatus: Option[PaymentStatus]): Amendment = Amendment(
+    LocalDateTime.now,
+    DeclarationGoods(createImport(desc, "99.99") :: Nil),
+    None,
+    paymentStatus,
+    Some("Digital")
+  )
```

Comment:
i think we already have all those data from the `CoreTestData` class ?  you can just reuse them.. if not move these into the class